src/vm/memory/memory.zig

```diff
@@ -762,6 +762,46 @@ pub const Memory = struct {
         }
         return values;
     }
+
+    /// Retrieves a continuous range of `Felt252` values starting from a specified address.
+    ///
+    /// # Arguments
+    ///
+    /// * `address`: The starting address in the memory from which the continuous range of `Felt252` is retrieved.
+    /// * `size`: The size of the continuous range of `Felt252` to be retrieved.
+    ///
+    /// # Returns
+    ///
+    /// Returns a list containing `Felt252` values retrieved from the continuous range starting at the given address.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if there are any gaps encountered within the continuous memory range.
+    /// Returns an error if value inside the range is not a `Felt252`
+    pub fn getFeltRange(
+        self: *Self,
+        address: Relocatable,
+        size: usize,
+    ) !std.ArrayList(Felt252) {
+        var values = try std.ArrayList(Felt252).initCapacity(
+            self.allocator,
+            size,
+        );
+        errdefer values.deinit();
+        for (0..size) |i| {
+            const elem = self.getFelt(try address.addUint(@intCast(i))) catch |err| {
+                // getFelt return ExpectedInteger
+                if (err == error.ExpectedInteger) {
+                    if (null == try (self.get(try address.addUint(@intCast(i))))) {
+                        return MemoryError.GetRangeMemoryGap;
+                    }
+                }
+                return err;
+            };
+            try values.append(elem);
```

Comment:
Here why do we need specific error modifications? I think that doing:

```zig
try values.append(try self.getFelt(try address.addUint(@intCast(i))));
```

is enough because we must not modify the errors returned by the `getFelt` method. I just saw that the errors in the `getFelt` method are not exactly correct, it is better that you modify them directly in the `getFelt` method itself which should look like:

```zig
pub fn getFelt(
         self: *Self,
         address: Relocatable,
     ) error{ MemoryOutOfBounds, ExpectedInteger }!Felt252 {
         if (try self.get(address)) |m| {
             return switch(m) {
                 .felt => |fe| fe,
                 else => MemoryError.ExpectedInteger,
             };
         } else {
             return MemoryError.UnknownMemoryCell;
         }
     }
```

You should then add these errors in `MemoryError`:

https://github.com/keep-starknet-strange/cairo-zig/blob/18dccfa07e7f1211c821b4502121af5cfd1e0e16/src/vm/error.zig#L37-L56

and maybe adjust some unit tests.

Reply:
I have added `ExpectedInteger` and `UnknownMemoryCell` in `MemoryError` but I have to remove error set in `getFelt` since I didn't find a way to do a declaration like:
```zig
error { MemoryOutBounds} || MemoryError { ExpectedInteger, UnknownMemoryCell} ! Felt252
```

Test cases for `getFelt` have been updated.

Reply:
Yes no problem fine :) This seems good to me, pending @lana-shanghai green light for merging and conflict resolution with main branch.

Reply:
I have looked at conflict and I'm wondering if I should keep `MemoryError.ExpectedInteger` and `MemoryError.UnknowMemoryCell` error for `getFelt` or update `getFeltRange` to use `error.ExpectedInteger` and `error.MemoryOutOfBounds`

@tcoratger what do you think ?

Reply:
@ptisserand For me we should stay with:

```zig
pub fn getFelt(
         self: *Self,
         address: Relocatable,
     ) error{ MemoryOutOfBounds, ExpectedInteger }!Felt252 {
         if (try self.get(address)) |m| {
             return switch(m) {
                 .felt => |fe| fe,
                 else => MemoryError.ExpectedInteger,
             };
         } else {
             return MemoryError.UnknownMemoryCell;
         }
     }
```

and not modify this error in `getFeltRange` which should inherit from the errors of `getFelt`

Reply:
Ok, I have fixed conflict and rebased on main.